### PR TITLE
[#11085] Instructor edit questions: tweak option order for generating answers from teams

### LINKED ISSUE #11085: Instructor edit questions: tweak option order for generating answers from teams
v7.13.0

image

Shouldn't these two be together?

### PR DESCRIPTION
**Pull request by tlhuynh**
Fixes #11085 

**Outline of Solution**

The goal is to reorder the list from "Or, generate options for the list of all" checkbox for the multiple-choice type questions.

The fix was to reorder the strings inside PARTICIPANT_TYPES to match with the expected order.

The files edited were:

src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts

src/web/app/components/question-types/question-edit-details-form/msq-question-edit-details-form.component.ts

### CODE REVIEW

**Review by daongochieu2810 — CHANGES_REQUESTED** [review]
Changes look good to me! However please remove changes made in `.vscode/settings.json`, `angular.json` and the alarmingly large `package-lock,.json`

**Comment by moziliar** [comment]
Do remove file `.vscode/settings.json` (I suppose you should add it in your `.gitignore`, preferably globally too), and `git restore` both `angular.json` and `package-lock.json`, which were a result of your npm install.

**Comment by tlhuynh** [comment]
Alright! I made the changes. I'll keep this in mind for the next PR. Regarding the package-lock.json and angular.json, what do you guys normally do to avoid big changes added to those files due to npm install?

**Comment by moziliar** [comment]
@tlhuynh Before you commit, you should check your `git status` (and even `git diff`) to ensure that you are going to commit the intended changes; if you find changes to files like `package-lock.json`, you could just `git restore` it, since the file is about locking the dependency versions in your `package.json` and you shouldn't really expect it to be different from the master unless you add in new dependency (and lock again). For changes to `angular.json`, you could just simply omit it when committing.

Of course, the cause of the additional line in `angular.json` is that you enabled angular's analytics by default (becoz Google); you can do `ng analytics off` to manually turn it off, or just indicate N when prompted upon `nom run start` (which in turn runs `ng serve` that normally prompts you so anyways).

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 .DS_Store
+.vscode/
```
Comment (moziliar) [inline .gitignore:2]: Not sure if we should explicitly add this here.. Ideally, every dev should kind of add it anyways.
@rrtheonlyone should we add vsc gitignore in this PR? I see that IDEA is added but not vsc.

File: src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts [hunk 2]
```diff
@@ -16,9 +16,12 @@ export class McqQuestionEditDetailsFormComponent
     extends QuestionEditDetailsFormComponent<FeedbackMcqQuestionDetails> {
 
   readonly PARTICIPANT_TYPES: string[] = [FeedbackParticipantType.STUDENTS,
```
Comment (moziliar) [inline src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts:18]: while you are at it, maybe add line break before the first element in the array too?

File: src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts [hunk 3]
```diff
@@ -16,9 +16,12 @@ export class McqQuestionEditDetailsFormComponent
...
-    FeedbackParticipantType.STUDENTS_EXCLUDING_SELF, FeedbackParticipantType.TEAMS,
-    FeedbackParticipantType.OWN_TEAM_MEMBERS_INCLUDING_SELF, FeedbackParticipantType.OWN_TEAM_MEMBERS,
-    FeedbackParticipantType.TEAMS_EXCLUDING_SELF, FeedbackParticipantType.INSTRUCTORS];
+    FeedbackParticipantType.STUDENTS_EXCLUDING_SELF,
+    FeedbackParticipantType.TEAMS,
+    FeedbackParticipantType.TEAMS_EXCLUDING_SELF,
+    FeedbackParticipantType.OWN_TEAM_MEMBERS_INCLUDING_SELF,
+    FeedbackParticipantType.OWN_TEAM_MEMBERS,
+    FeedbackParticipantType.INSTRUCTORS];
```
Comment (moziliar) [inline src/web/app/components/question-types/question-edit-details-form/mcq-question-edit-details-form.component.ts:24]: line break before `]`

File: src/web/app/components/question-types/question-edit-details-form/msq-question-edit-details-form.component.ts [hunk 4]
```diff
@@ -20,9 +20,12 @@ export class MsqQuestionEditDetailsFormComponent
...
   readonly PARTICIPANT_TYPES: string[] = [FeedbackParticipantType.STUDENTS,
-    FeedbackParticipantType.STUDENTS_EXCLUDING_SELF, FeedbackParticipantType.TEAMS,
-    FeedbackParticipantType.OWN_TEAM_MEMBERS_INCLUDING_SELF, FeedbackParticipantType.OWN_TEAM_MEMBERS,
-    FeedbackParticipantType.TEAMS_EXCLUDING_SELF, FeedbackParticipantType.INSTRUCTORS];
+    FeedbackParticipantType.STUDENTS_EXCLUDING_SELF,
+    FeedbackParticipantType.TEAMS,
+    FeedbackParticipantType.TEAMS_EXCLUDING_SELF,
+    FeedbackParticipantType.OWN_TEAM_MEMBERS_INCLUDING_SELF,
+    FeedbackParticipantType.OWN_TEAM_MEMBERS,
+    FeedbackParticipantType.INSTRUCTORS];
```
Comment (moziliar) [inline src/web/app/components/question-types/question-edit-details-form/msq-question-edit-details-form.component.ts:28]: ditto.

File: src/web/app/components/question-types/question-edit-details-form/msq-question-edit-details-form.component.ts [hunk 5]
```diff
@@ -20,9 +20,12 @@ export class MsqQuestionEditDetailsFormComponent
     extends QuestionEditDetailsFormComponent<FeedbackMsqQuestionDetails> {
 
   readonly PARTICIPANT_TYPES: string[] = [FeedbackParticipantType.STUDENTS,
```
Comment (moziliar) [inline src/web/app/components/question-types/question-edit-details-form/msq-question-edit-details-form.component.ts:22]: ditto

**Review by moziliar — CHANGES_REQUESTED** [review]
Maybe you should reorder the options further: 

student -> student exclude self
team -> team exclude self
**own team exclude self -> own team** (reversed?)

**Comment by tlhuynh** [comment]
@moziliar I see I see. Thank you so much. I'll make sure to pay attention to this more from now on.

File: .gitignore (hunk 1 above)
Comment (moziliar) [reply .gitignore:2]: I think for now, you could try to add this line to your own global `.gitignore` first instead of project local.

**Comment by moziliar** [comment]
One thing to note is that for UI changes, it is best to attach screenshot of all the affected components.

**Comment by tlhuynh** [comment]
Do you mean for the PR description? Like screenshots for how it was before and how it looks after the changes? I thought about it but wasn't sure if it was necessary. Will do, if I work with UIs again.

File: .gitignore (hunk 1 above)
Comment (tlhuynh) [reply .gitignore:2]: Got it. I will get to it now.

**Review by moziliar — APPROVED** [review]
LGTM. @damithc do you want to vet the ordering of the options too?

**Comment by damithc** [comment]
> @damithc do you want to vet the ordering of the options too?

Yes please. Let me know when a screenshot has been posted.

**Comment by moziliar** [comment]
@tlhuynh typically, a screenshot is needed for UI related changes. Do attach screenshots for both dropdown (I see two are changed and only these two are needed)

**Comment by tlhuynh** [comment]
@damithc , @moziliar 

Here are the screenshots for the changes:

This one is for the MC-single answer
Multiple-choice Single Answer

And, this one is for the MC-multi answers
Multiple-choice Multiple Answers

**Comment by damithc** [comment]
> his one is for the MC-single answer
> Multiple-choice Single Answer

Not what I was expecting when I posted the original issue. Isn't it easier for the user if we pair `x` and x `(excluding ...)`? 

e.g.,
```
students
students (excluding self)
...
```

**Comment by tlhuynh** [comment]
@damithc 
That was the original fix, but @moziliar wants to see if we can reorder it further and put them into groups (with self and excluding self). I can always reorder them depend on how you guys want them.

**Comment by moziliar** [comment]
@tlhuynh I didn't imply grouping by the way; instead, I was just pointing out that the `self` and `excluding self` ordering of the last pair was reversed.

**Comment by tlhuynh** [comment]
@moziliar 
Ahh, I see I see. I misunderstood, then. I thought you wanted to separate them. That was my fault. Making the change now.

**Comment by tlhuynh** [comment]
@damithc 
Here are the new screenshots:
For MC-single answer:
New-MC-single answer

For MC-multi answers:
New-MC-multi answers

**Comment by damithc** [comment]
> For MC-single answer:
> New-MC-single answer

Looks good to me 👍

**Review by daongochieu2810 — APPROVED** [review]
lgtm

**Pull request merged**